### PR TITLE
TRUNK-5809: Program Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by alinmihaila99**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5809

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -82,7 +82,6 @@
 		<mapping resource="org/openmrs/api/db/hibernate/EncounterType.hbm.xml" />
         <mapping resource="org/openmrs/api/db/hibernate/EncounterProvider.hbm.xml" />
 		<mapping resource="org/openmrs/api/db/hibernate/EncounterRole.hbm.xml" />
-		<mapping resource="org/openmrs/api/db/hibernate/Program.hbm.xml" />
                 <mapping resource="org/openmrs/api/db/hibernate/ProgramAttributeType.hbm.xml" />
```
Comment (HerbertYiga) [inline api/src/main/resources/hibernate.cfg.xml:85]: we could also have these mapping files on the same line

Comment (alinmihaila99) [reply api/src/main/resources/hibernate.cfg.xml:85]: Yes I saw this, but I didn't know if it's ok to modify something that is not in my ticket

Comment (HerbertYiga) [reply api/src/main/resources/hibernate.cfg.xml:85]: it is ok to align that

File: api/src/main/java/org/openmrs/Program.java [hunk 2]
```diff
@@ -13,30 +13,64 @@
...
 	// ******************
 	// Properties
 	// ******************
-	
+
+	@Id
+	@Column(name = "program_id", nullable = false)
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "program_id_gen")
+	@SequenceGenerator(name = "program_id_gen", sequenceName = "program_program_id_seq")
 	private Integer programId;
```
Comment (Ruhanga) [inline api/src/main/java/org/openmrs/Program.java:55]: @alinmihaila99, could you not plainly use `GenerationType.IDENTITY`?

**Review by Ruhanga — COMMENTED** [review]
Remember to resolve merge conflicts on this PR.

**Review by tendomart — APPROVED** [review]

**Review by tendomart — CHANGES_REQUESTED** [review]
Was it necessary to consider the Travis log failure ?

**Pull request closed without merging**